src/main/java/org/broadinstitute/hellbender/engine/spark/JoinReadsWithVariants.java

```diff
@@ -66,7 +66,16 @@
 
         JavaPairRDD<GATKRead, Variant> allPairs = pairReadsWithVariants(readsWShards, variantsWShards);
 
-        return allPairs.distinct().groupByKey();
+        return allPairs.aggregateByKey(new HashSet<>(), (vs, v) -> {
+            if (v != null) { // pairReadsWithVariants can produce null variant
+                ((HashSet<Variant>) vs).add(v);
```

Comment:
Can you avoid the typecasting by adding explicit types for the lamda parameters?